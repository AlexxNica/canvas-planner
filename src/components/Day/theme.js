/* Global variables (colors, typography, spacing, etc.) are defined in lib/themes */

export default function generator ({ colors, typography, spacing }) {
  return {
    fontSize: typography.fontSizeMedium,
    fontFamily: typography.fontFamily,
    fontWeight: typography.fontWeightNormal,
    lineHeight: typography.lineHeightCondensed,

    color: colors.oxford,
    background: colors.white,

    marginTop: spacing.large
  };
}
