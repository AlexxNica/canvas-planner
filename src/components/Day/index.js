import React, { Component } from 'react';
import themeable from 'instructure-ui/lib/themeable';
import Heading from 'instructure-ui/lib/components/Heading';
import Typography from 'instructure-ui/lib/components/Typography';
// import moment from 'moment';
import { string, arrayOf, object } from 'prop-types';
import styles from './styles.css';
import theme from './theme.js';
import { getFriendlyDate, getFullDate, isToday } from '../../utilities/dateUtils';
import { groupBy } from 'lodash';


class Day extends Component {
  static propTypes = {
    day: string.isRequired,
    itemsForDay: arrayOf(object)
  }

  constructor (props) {
    super(props);

    this.friendlyName = getFriendlyDate(props.day);
    this.fullDate = getFullDate(props.day);
    this.state = {
      groupedItems: this.groupItems(props.itemsForDay)
    };
  }

  componentWillReceiveProps (nextProps) {
    this.setState((state) => {
      return {
        groupedItems: this.groupItems(nextProps.itemsForDay)
      };
    });
  }

  groupItems = (items) => groupBy(items, item => (item.context && item.context.id) || 'Notes');

  render () {
    return (
      <div className={styles.root}>
          <Heading>
            <Typography
              as="div"
              transform="uppercase"
              lineHeight="condensed"
              size={isToday(this.props.day) ? 'large' : 'medium'}
            >
              {this.friendlyName}
            </Typography>
            <Typography
              as="div"
              lineHeight="condensed"
            >
              {this.fullDate}
            </Typography>
          </Heading>

        <div className={styles.dayContent}>
          {
            Object.keys(this.state.groupedItems).map((cid) => {
              return (<div key={cid}>Grouping Placeholder for course id:{cid}</div>);
            })
          }
        </div>
      </div>
    );
  }
}

export default themeable(theme, styles)(Day);
