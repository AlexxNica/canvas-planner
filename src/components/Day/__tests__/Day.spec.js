import React from 'react';
import 'instructure-ui/lib/themes/canvas';
import { shallow } from 'enzyme';
import moment from 'moment';
import Day from '../index';

it('renders the base component with required props', () => {
  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the friendly name in large text when it is today', () => {
  const today = moment();

  const wrapper = shallow(
    <Day timeZone="America/Denver" day={today.format('YYYY-MM-DD')} />
  );
  expect(wrapper.find('Typography').first().props().size).toEqual('large');
});

it('renders the friendlyName in medium text when it is not today', () => {
  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" />
  );
  expect(wrapper.find('Typography').first().props().size).toEqual('medium');
});

it('groups itemsForDay based on context id', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      id: 128
    }
  }, {
    title: 'San Juan',
    context: {
      id: 256
    }
  }, {
    title: 'Roll for the Galaxy',
    context: {
      id: 256
    }
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  const groupedItems = wrapper.state('groupedItems');
  expect(groupedItems[128].length).toEqual(1);
  expect(groupedItems[256].length).toEqual(2);
});

it('groups itemsForDay that have no context into the "Notes" category', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      id: 128
    }
  }, {
    title: 'San Juan',
    context: {
      id: 256
    }
  }, {
    title: 'Roll for the Galaxy',
    context: {
      id: 256
    }
  }, {
    title: 'Get work done!'
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  const groupedItems = wrapper.state('groupedItems');
  expect(groupedItems.Notes.length).toEqual(1);
});

it('groups itemsForDay that come in on prop changes', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      id: 128
    }
  }, {
    title: 'San Juan',
    context: {
      id: 256
    }
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  let groupedItems = wrapper.state('groupedItems');
  expect(Object.keys(groupedItems).length).toEqual(2);

  const newItemsForDay = items.concat([{
    title: 'Roll for the Galaxy',
    context: {
      id: 256
    }
  }, {
    title: 'Get work done!'
  }]);

  wrapper.setProps({ itemsForDay: newItemsForDay });
  groupedItems = wrapper.state('groupedItems');
  expect(Object.keys(groupedItems).length).toEqual(3);
});
