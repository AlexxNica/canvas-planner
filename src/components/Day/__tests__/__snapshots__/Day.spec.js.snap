// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the base component with required props 1`] = `
<div>
  <Heading
    border="none"
    color="inherit"
    ellipsis={false}
    level="h2"
  >
    <Typography
      as="div"
      lineHeight="condensed"
      size="medium"
      transform="uppercase"
    >
      Yesterday
    </Typography>
    <Typography
      as="div"
      lineHeight="condensed"
      size="medium"
    >
      Tuesday, April 25
    </Typography>
  </Heading>
  <div />
</div>
`;
